src/goto-checker/goto_trace_storage.cpp

```diff
@@ -17,14 +17,24 @@ goto_trace_storaget::goto_trace_storaget(const namespacet &ns) : ns(ns)
 
 const goto_tracet &goto_trace_storaget::insert(goto_tracet &&trace)
 {
-  traces.push_back(trace);
+  traces.push_back(std::move(trace));
   const auto &last_step = traces.back().get_last_step();
   DATA_INVARIANT(
     last_step.is_assert(), "last goto trace step expected to be assertion");
   property_id_to_trace_index.emplace(last_step.property_id, traces.size() - 1);
   return traces.back();
 }
 
+const goto_tracet &goto_trace_storaget::insert_all(goto_tracet &&trace)
+{
+  traces.push_back(std::move(trace));
+  for(const auto &property_id : traces.back().get_all_property_ids())
```

Comment:
Do you expect `insert_all` to ever be called on a trace hitting no properties? If not, can you add an invariant that `traces.back().get_all_property_ids()` is non-empty?

Reply:
There wouldn't be a trace of no property was hit, but right, this invariant is quite far away from this place. I'll add it.